File: codes/pythagorean-triple.py
def valid(i,j,k):
    """
    >>> valid(3,4,5)
    True
    >>> valid(5,4,3)
    False
    """
    return (k**2 == i**2 + j**2)

n = 20
r = range(1,n)

ans = [(x,y,z) for x in r for y in r for z in r if valid(x,y,z)]
for i in ans:
    print(i)

# for x in r:
#     for y in r:
#         for z in r:
#             if valid(x,y,z):
#                 print (x,y,z)

File: codes/pythagorean-triple.hs
module PTriple where

satisfy :: Int -> Int -> Int -> Bool
satisfy i j k = k^2 == i^2 + j^2

ans :: Int -> [(Int,Int,Int)]
ans n = [(a,b,c) | a <- [1..n], b <- [1..n], c <- [1..n], satisfy a b c]
